docs/rfcs/0002-layer-architecture.md

```diff
@@ -0,0 +1,328 @@
+
+<a href="0002-layer-architecture.pdf" target="_blank" rel="noopener" class="print-pdf">Download as PDF</a> (for docs.racklet.io)
+
+[TOC]
+
+## RFC Metadata
+
+**Authors** (in alphabetical order):
+
+- Dennis Marttinen, [@twelho](https://github.com/twelho)
+- Lucas Käldström, [@luxas](https://github.com/luxas)
+
+**Status** (as defined [here]): `Provisional`
+
+[here]: https://github.com/kubernetes/enhancements/blob/46b95331113de10ca20a11eb41a0131c88bca966/keps/NNNN-kep-template/kep.yaml#L9
+
+**Creation Date**: `2021-04-15`
+
+**Last Updated**: `2021-06-11`
+
+**RFC Handle**: `0002-layer-architecture`
+
+**Initial Pull Request**: [racklet/racklet#20](https://github.com/racklet/racklet/pull/20)
+
+**Tracking Issue**: [racklet/racklet#21](https://github.com/racklet/racklet/issues/21)
+
+## Summary
+
+_One paragraph explanation of the feature._
+
+This RFC describes the overall Racklet architecture, its defining layers, and requirements for each such layer, derived from [RFC-0001]. For each layer the defining components are described at a high level (avoiding implementation details). The compnoents are associated with their role and five highlighted key requirements from the values and user goals of [RFC-0001].
+
+[RFC-0001]: 0001-high-level-architecture.md
+
+## Motivation
+
+_Why are we doing this? What use cases does it support? What is the expected outcome?_
+
+With this RFC we aim to clearly define the layers Racklet consists of to provide a clear overview of the system for all contributors and maintainers. Additionally this document concisely presents the techniques and technologies used in the various layers to achieve the goals stated in [RFC-0001]. The textual and tabular representation provided here is expected to be augmented by graphical visualizations of the architecture.
+
+**TODO**: Embed the graphics into this document when they're ready.
+
+### Goals
+
+_What is in scope for this work?_
+
+- Define well-known layers of Racklet.
+- Describe the requirements for each layer.
+- Describe roughly how to be "Racklet conformant" and what the differences are between Racklet and other similar alternatives.
+
+### Non-Goals
+
+_What is out of scope for this work?_
+
+- Describe the details and/or technical implementations of the various layers. See the detailed RFCs for the layers if looking for that information.
+- Cover every minor component or implementation-specific components, this RFC is designed to only give an overview.
+
+## Proposal
+
+_This is the technical portion of the RFC. Explain the design in sufficient detail that:_
+
+- _Its interaction with other features is clear._
+- _It is reasonably clear how the feature would be implemented._
+- _Corner cases are dissected by example._
+
+_The section should return to the examples given in the guide-level explanation below, and explain more fully how the detailed proposal makes those examples work._
+
+Racklet is divided into 5 distinct layers, from lowest-level to highest-level:
```

Comment:
reverse this so it's in line with how the RFC is read?

Reply:
Yep, Markdown was being a bit difficult with reversed lists, but I managed to find a workaround.